Build fails with shared ON

Hi,

Thanks for this nice project which is very useful for those of us using CMake. The static version of the project built OK, although I had to remove on line since I am using gcc-6.3 (on Debian stable):
```

diff --git a/tensorflow_cc/cmake/TensorflowShared.cmake b/tensorflow_cc/cmake/TensorflowShared.cmake
index f1e9d55..d6cdb1d 100644
--- a/tensorflow_cc/cmake/TensorflowShared.cmake
+++ b/tensorflow_cc/cmake/TensorflowShared.cmake
@@ -12,7 +12,6 @@ ExternalProject_Add(
   UPDATE_COMMAND ""
   CONFIGURE_COMMAND tensorflow/contrib/makefile/compile_linux_protobuf.sh
             # patch nsync to use g++-5
-            COMMAND sed -i "s/ g++/ g++-5/g" tensorflow/contrib/makefile/compile_nsync.sh
             COMMAND tensorflow/contrib/makefile/compile_nsync.sh
             COMMAND cp "${CMAKE_CURRENT_SOURCE_DIR}/cmake/build_tensorflow.sh" .
             COMMAND cp "${CMAKE_CURRENT_SOURCE_DIR}/cmake/copy_links.sh" .
```
After that and I was able to use the example to check that everything is OK.

However, the shared version fails with this error:

```
compile_linux_protobuf.sh finished successfully!!!
+ g++ -DNSYNC_USE_CPP11_TIMEPOINT -DNSYNC_ATOMIC_CPP11 -I../../platform/c++11.futex -I../../platform/c++11 -I../../platform/gcc -I../../platform/posix -pthread -I../../public -I../../internal -O -std=c++11 -Werror -Wall -Wextra -pedantic -c ../../platform/linux/src/nsync_semaphore_futex.c
+ g++ -DNSYNC_USE_CPP11_TIMEPOINT -DNSYNC_ATOMIC_CPP11 -I../../platform/c++11.futex -I../../platform/c++11 -I../../platform/gcc -I../../platform/posix -pthread -I../../public -I../../internal -O -std=c++11 -Werror -Wall -Wextra -pedantic -c ../../platform/c++11/src/per_thread_waiter.cc
+ g++ -DNSYNC_USE_CPP11_TIMEPOINT -DNSYNC_ATOMIC_CPP11 -I../../platform/c++11.futex -I../../platform/c++11 -I../../platform/gcc -I../../platform/posix -pthread -I../../public -I../../internal -O -std=c++11 -Werror -Wall -Wextra -pedantic -c ../../platform/c++11/src/yield.cc
+ g++ -DNSYNC_USE_CPP11_TIMEPOINT -DNSYNC_ATOMIC_CPP11 -I../../platform/c++11.futex -I../../platform/c++11 -I../../platform/gcc -I../../platform/posix -pthread -I../../public -I../../internal -O -std=c++11 -Werror -Wall -Wextra -pedantic -c ../../platform/c++11/src/time_rep_timespec.cc
+ g++ -DNSYNC_USE_CPP11_TIMEPOINT -DNSYNC_ATOMIC_CPP11 -I../../platform/c++11.futex -I../../platform/c++11 -I../../platform/gcc -I../../platform/posix -pthread -I../../public -I../../internal -O -std=c++11 -Werror -Wall -Wextra -pedantic -c ../../platform/c++11/src/nsync_panic.cc
tensorflow/contrib/makefile/downloads/nsync/builds/default.linux.c++11/nsync.a
CUDA support disabled
You have bazel 0.13.1- (@non-git) installed.
Preconfigured Bazel build configs. You can use any of the below by adding "--config=<>" to your build command. See tools/bazel.rc for more details.
	--config=mkl         	# Build with MKL support.
	--config=monolithic  	# Config for mostly static monolithic build.
Configuration finished
Starting local Bazel server and connecting to it...
..........
WARNING: /home/user/.cache/bazel/_bazel_user/6f6dfefb4e43ac28cb8d73e15d94e2f8/external/protobuf_archive/WORKSPACE:1: Workspace name in /home/user/.cache/bazel/_bazel_user/6f6dfefb4e43ac28cb8d73e15d94e2f8/external/protobuf_archive/WORKSPACE (@com_google_protobuf) does not match the name given in the repository's definition (@protobuf_archive); this will cause a build error in future versions
INFO: Analysed target //tensorflow:libtensorflow_cc.so (96 packages loaded).
INFO: Found 1 target...
Target //tensorflow:libtensorflow_cc.so up-to-date:
  bazel-bin/tensorflow/libtensorflow_cc.so
INFO: Elapsed time: 62.900s, Critical Path: 14.75s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
Replacing links with the copies of their targets.
This may take a while...
rm: cannot remove './bazel-tensorflow/external/local_jdk/lib/jexec': Permission denied
CMakeFiles/tensorflow_shared.dir/build.make:106: recipe for target 'tensorflow-stamp/tensorflow_shared-configure' failed
make[2]: *** [tensorflow-stamp/tensorflow_shared-configure] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/tensorflow_shared.dir/all' failed
make[1]: *** [CMakeFiles/tensorflow_shared.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

```

Thanks.

The ./bazel-tensorflow dir does not exist:

```
Replacing links with the copies of their targets.
This may take a while...
rm: cannot remove './bazel-tensorflow/external/local_jdk/lib/jexec': Permission denied
CMakeFiles/tensorflow_shared.dir/build.make:106: recipe for target 'tensorflow-stamp/tensorflow_shared-configure' failed
make[2]: *** [tensorflow-stamp/tensorflow_shared-configure] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/tensorflow_shared.dir/all' failed
make[1]: *** [CMakeFiles/tensorflow_shared.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
You have new mail in /var/mail/user
07:37:12-user~/local/src/tensorflow_cc/tensorflow_cc/build (master)$ ls -la ./bazel-tensorflow/external/local_jdk/lib/jexec
ls: cannot access './bazel-tensorflow/external/local_jdk/lib/jexec': No such file or directory
07:39:27-user~/local/src/tensorflow_cc/tensorflow_cc/build (master)$ ls -la ./bazel-tensorflow/external/local_jdk/lib/
ls: cannot access './bazel-tensorflow/external/local_jdk/lib/': No such file or directory
07:39:44-user~/local/src/tensorflow_cc/tensorflow_cc/build (master)$ ls -la ./bazel-tensorflow/external/local_jdk/
ls: cannot access './bazel-tensorflow/external/local_jdk/': No such file or directory
07:39:46-user~/local/src/tensorflow_cc/tensorflow_cc/build (master)$ ls -la ./bazel-tensorflow/external/
ls: cannot access './bazel-tensorflow/external/': No such file or directory
07:39:51-user~/local/src/tensorflow_cc/tensorflow_cc/build (master)$ ls -la ./bazel-tensorflow/
ls: cannot access './bazel-tensorflow/': No such file or directory
07:39:56-user~/local/src/tensorflow_cc/tensorflow_cc/build (master)$ ls -la
total 64
drwxr-xr-x 7 user user  4096 Jun  5 07:22 .
drwxr-xr-x 5 user user  4096 May 31 11:10 ..
-rw-r--r-- 1 user user 12233 Jun  5 07:21 CMakeCache.txt
drwxr-xr-x 8 user user  4096 Jun  5 07:23 CMakeFiles
-rw-r--r-- 1 user user  4891 Jun  5 07:21 cmake_install.cmake
-rw-r--r-- 1 user user  6463 Jun  5 07:21 Makefile
drwxr-xr-x 8 user user  4096 Jun  5 07:36 tensorflow
drwxr-xr-x 2 user user  4096 Jun  5 07:21 tensorflow_base-prefix
-rw-r--r-- 1 user user   803 Jun  5 07:21 TensorflowCCConfig.cmake
-rw-r--r-- 1 user user  1607 Jun  5 07:21 TensorflowCCConfigVersion.cmake
drwxr-xr-x 3 user user  4096 Jun  5 07:21 tensorflow_shared-prefix
drwxr-xr-x 2 user user  4096 Jun  5 07:23 tensorflow-stamp
```

--- a/tensorflow_cc/cmake/copy_links.sh
+++ b/tensorflow_cc/cmake/copy_links.sh
@@ -14,7 +14,7 @@ find -L "$1" -depth -print0 |
         # the original when some of its parent directories were copied.
         # Also the first check is to detect whether the file (after
         # symlink dereference) exists so that `realpath` does not fail.
-        if [[ -e "$f" ]] && [[ -L "$f" ]]; then
+        if [[ -w "$f" ]] && [[ -L "$f" ]]; then
             realf="$(realpath "$f")"
             rm "$f"
             cp -r --link "$realf" "$f" 2>/dev/null || cp -r "$realf" "$f"

